### PR TITLE
Extract position getter and setters to a facet

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
In order to improve build times as revealed by #41237, This provides an example of faceting the creature.character/player/avatar classes, extracting a portion of the class interface so it can be included more cheaply.

#### Describe the solution
Create a pure virtual class location to declare the pos(), pos[xyz]() and setpos() methods so that just the location associated with a creature can be passed as an argument or retrieved with a global getter without including the heavier creature or character headers.
Also added a global getter for retrieving the avatar instance as a location and used it.

#### Describe alternatives you've considered
There are a number of similar options around shrinking the character header (the main offender), but after auditing the retrievals of especially the avatar instance, I think there's a lot of low hanging fruit in this area that would be even better than the minimal character implementation.

#### Testing
Everything builds, accessors clearly work and are sufficiently exercised by various tests.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
It looks like you didn't add location.h

**Comment by kevingranade** [comment]
Pfff tiny details

File: src/creature.h [hunk 1]
```diff
@@ -196,7 +197,7 @@ enum class FacingDirection : int {
     RIGHT = 2
 };
 
-class Creature
+class Creature : public location
 {
     public:
         virtual ~Creature();
```
Comment (kevingranade) [inline src/creature.h:203]:
```suggestion
        ~Creature() override;
```

**Comment by kevingranade** [comment]
This won't be sufficient to make much of a dent, but I think it's a worthwhile facet to have.

**Pull request merged**